content/general/concepts/data-structures/terms/algorithms/searching/terms/binary-search/binary-search.md

```diff
@@ -0,0 +1,69 @@
+---
+Title: 'Binary Search Algorithm'
+Description: 'A searching algorithm to search an element in a sorted collection by dividing the collection in two halves.'
+Subjects:
+  - 'Computer Science'
+  - 'Code Foundations'
+Tags:
+  - 'Algorithms'
+  - 'Data Structures'
+  - 'Search Algorithms'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+**Binary Search** is an algorithm for searching an element within a sorted collection of items, primarily implemented with [arrays](https://www.codecademy.com/resources/docs/general/data-structures/array) or lists. The binary search algorithm follows a divide-and-conquer approach to determine whether the target element matches the collection's middle element. If there is a match, it provides the index of the middle element; otherwise, it proceeds to either side of the array, depending on the current comparison result.
+
+> **Note**: The collection must be sorted for the implementation of the binary search algorithm.
+
+## The Algorithm
+
+The steps for the binary search algorithm are as follows: 
+
+1. Set the `start` pointer to the beginning of the collection (index 0).
+2. Set the `end` pointer to the end of the collection (length(collection) - 1).
+3. While the `start` is less than or equal to the `end` pointer, repeat these steps:
+    1. Calculate the middle element index: `mid = start + (end - start) / 2`.
+    2. Compare the value at middle index (`mid`) with the target value.
+        1. If `arr[mid]` is equal to the target value, return `mid` (search successful).
+        2. If `arr[mid]` is less than the target value, set the `start` to `mid + 1`.
+        3. If `arr[mid]` is greater than the target value, set the `end` to `mid - 1`.
+4. If the `start` pointer becomes greater than the `end` pointer, the target value is not in the collection. Return `-1` to indicate that the target is not present.
+
+## Complexities for Binary Search Algorithm
+
+### Time Complexity
+
+- Average Case: `O(log n)`
+- Worst Case: `O(log n)`
+- Best Case: `O(1)`
+
+The binary search algorithm has logarithmic time complexity because it divides the array repeatedly until the target element is discovered or the search space is empty.
```

Comment:
There should be a mention that worst case is determined by the algorithm searching for an element that does not exist in the collection.